NExtends/Primitives/Strings/String.extensions.cs

```diff
@@ -468,33 +468,38 @@ public static string[] ToLower(this string[] array)
             return array.Select(s => s.ToLower()).ToArray();
         }
 
+#if !NETCOREAPP2_0
         /// <summary>
-        /// String replacement with case insensivity support
-        /// https://stackoverflow.com/questions/6275980/string-replace-ignoring-case
+        /// Replace String in net461
         /// </summary>
-        /// <param name="originalString"></param>
-        /// <param name="oldValue">The string to be replaced.</param>
-        /// <param name="newValue">The string to replace all occurrences of oldValue.</param>
-        /// <param name="comparisonType"></param>
-        /// <returns></returns>
         public static string Replace(this string originalString, string oldValue, string newValue, StringComparison comparisonType)
         {
-            var regexOptions = RegexOptions.IgnoreCase;
+            if (oldValue == null)
+                throw new ArgumentNullException("oldValue");
+            if (oldValue.Length == 0)
+                throw new ArgumentException("String cannot be of zero length.", "oldValue");
+
+            StringBuilder sb = null;
 
-            if (comparisonType == StringComparison.CurrentCulture
-                || comparisonType == StringComparison.InvariantCulture
-                || comparisonType == StringComparison.Ordinal)
+            int startIndex = 0;
+            int foundIndex = originalString.IndexOf(oldValue, comparisonType);
+            while (foundIndex != -1)
             {
-                regexOptions = RegexOptions.CultureInvariant;
+                if (sb == null)
+                    sb = new StringBuilder(originalString.Length + (newValue != null ? Math.Max(0, 5 * (newValue.Length - oldValue.Length)) : 0));
+                sb.Append(originalString, startIndex, foundIndex - startIndex);
+                sb.Append(newValue);
+
+                startIndex = foundIndex + oldValue.Length;
+                foundIndex = originalString.IndexOf(oldValue, startIndex, comparisonType);
             }
 
-            return Regex.Replace(
-                originalString,
-                Regex.Escape(oldValue),
-                newValue.Replace("$", "$$"),
-                regexOptions
-            );
+            if (startIndex == 0)
+                return originalString;
+            sb.Append(originalString, startIndex, originalString.Length - startIndex);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') 'sb' is null on at least one execution path. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS2259)